feat(ghes): fetch tool from configurable server

DariuszPorowski opened the pull request:
## 💌 Description

- Add option to set own GitHub REST API, default to GITHUB_API_URL
- Enforce actionlint version / matcher check from public github
- Inputs naming adjustments
- Update docs

## 🔗 Related issue

<!-- If your PR refers to a related issue, link it here. -->
Fixes: #14

## 🏗️ Type of change

<!-- Mark with an `x` all the checkboxes that apply (like `[x]`) -->

- [x] 📚 Examples/docs/tutorials
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🚨 Security fix
- [x] ⬆️ Dependencies update

## ✅ Checklist

<!-- Mark with an `x` all the checkboxes that apply (like `[x]`) -->

- [x] I've read the [`Code of Conduct`](https://github.com/raven-actions/actionlint/blob/main/.github/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`Contributing`](https://github.com/raven-actions/actionlint/blob/main/.github/CONTRIBUTING.md) guide.

bhundven-axon commented:
@DariuszPorowski I'll give this a shot in a little while and will update with results.

DariuszPorowski commented on action.yml:
```diff
@@ -86,7 +96,8 @@ runs:
       id: environment
       with:
-        github-token: ${{ inputs.token || env.GITHUB_TOKEN }}
+        github-token: ${{ inputs.github-token || inputs.token || env.GITHUB_TOKEN }}
```
@frederikb great input! thanks for checking. Going to address your suggestions today, and appreciate GHES test after will update this PR.

DariuszPorowski replied:
@frederikb the problem with anonymous GH public api calls is, it's a very limited rate limits: `The primary rate limit for unauthenticated requests is 60 requests per hour.`

DariuszPorowski committed:
feat(action): add github-server-url input for GHES support (fixes #14)

Adds a github-server-url input (default https://github.com) used to build the release lookup (redirect), binary download, matcher, and API fallback URLs. By default the external rhysd/actionlint tool is always fetched from public github.com, so the GHES API is never queried for it; GHES users who mirror rhysd/actionlint can point the action at their own server.

DariuszPorowski committed:
fix(inputs): update input table formatting and add github-server-url description

Signed-off-by: Dariusz Porowski <3431813+DariuszPorowski@users.noreply.github.com>

DariuszPorowski committed:
fix(cspell): add GHES and rhysd to custom words list

Signed-off-by: Dariusz Porowski <3431813+DariuszPorowski@users.noreply.github.com>
